src/common/services/youtube/search.ts

```diff
@@ -23,17 +21,16 @@ export const search: SearchFunction = async ({ artist, title }) => {
   const searchObjectString = [
     ...(/var ytInitialData = ({.*});/.exec(dataScript) ?? []),
   ][1]
-  if (isUndefined(searchObjectString))
-    throw new Error('Could not find search metadata')
+  if (!searchObjectString) throw new Error('Could not find search metadata')
 
-  const searchObject = decode(SearchObject)(searchObjectString)
+  const searchObject = JSON.parse(searchObjectString) as SearchObject
   const videoData =
     searchObject.contents.twoColumnSearchResultsRenderer.primaryContents
       .sectionListRenderer.contents[0]?.itemSectionRenderer?.contents
-  if (isUndefined(videoData)) throw new Error('Could not find video data')
+  if (!videoData) throw new Error('Could not find video data')
 
   const videoId = videoData
     .map((video) => video.videoRenderer)
-    .find(isDefined)?.videoId
+    .find((v) => v != undefined)?.videoId
```

Comment:
let's use `isDefined` here so we're consistent with using type guards for filter/find functions (at least until TS gets smarter about this stuff)